### PR TITLE
Issue #124 [PHPMD] - Set up PHPMD

### LINKED ISSUE #124: Setup PHP Mess Detector
In order to keep code quality good, we need to configure and run PHP Mess Detector

 - [x] Config 
    - More details + config generator can be found on http://edorian.github.io/php-coding-standard-generator/#phpmd
    - you can find my suggested config on the bottom of the issue
 - [x] Running PHP MD

```
./vendor/bin/phpmd src/ text phpmd.xml
```

 - [x] Fix issues in code

 - [x] Add PHPMD to CircleCI

Suggested config: `phpmd.xml`:

```
<?xml version="1.0" encoding="UTF-8"?>
<ruleset name="pcsg-generated-ruleset"
         xmlns="http://pmd.sf.net/ruleset/1.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://pmd.sf.net/ruleset/1.0.0 http://pmd.sf.net/ruleset_xml_schema.xsd"
         xsi:noNamespaceSchemaLocation="http://pmd.sf.net/ruleset_xml_schema.xsd">
    <description>Created with the PHP Coding Standard Generator.
        http://edorian.github.com/php-coding-standard-generator/
    </description>
    <exclude-pattern>cache/</exclude-pattern>
    <exclude-pattern>SymfonyRequirements.php</exclude-pattern>
    <exclude-pattern>AppKernel.php</exclude-pattern>
    <exclude-pattern>DoctrineMigrations/Version*.php</exclude-pattern>
    <exclude-pattern>DependencyInjection/Configuration.php</exclude-pattern>
    <exclude-pattern>DependencyInjection/*Extension.php</exclude-pattern>
    <exclude-pattern>src/AppBundle/Controller/DefaultController.php</exclude-pattern>
    <rule ref="rulesets/codesize.xml/CyclomaticComplexity"/>
    <rule ref="rulesets/codesize.xml/NPathComplexity"/>
    <rule ref="rulesets/codesize.xml/ExcessiveMethodLength"/>
    <rule ref="rulesets/codesize.xml/ExcessiveClassLength"/>
    <rule ref="rulesets/codesize.xml/ExcessiveParameterList"/>
    <rule ref="rulesets/codesize.xml/ExcessivePublicCount"/>
    <rule ref="rulesets/codesize.xml/TooManyFields">
        <properties>
            <property name="maxfields" value="25"/>
        </properties>
    </rule>
    <rule ref="rulesets/codesize.xml/TooManyMethods">
        <properties>
            <property name="maxmethods" value="40"/>
        </properties>
    </rule>
    <rule ref="rulesets/codesize.xml/ExcessiveClassComplexity">
        <properties>
            <property name="maximum" value="60"/>
        </properties>
    </rule>
    <rule ref="rulesets/controversial.xml"/>
    <rule ref="rulesets/design.xml"/>
    <rule ref="rulesets/naming.xml/ShortVariable">
        <properties>
            <property name="minimum" value="2"/>
        </properties>
    </rule>
    <rule ref="rulesets/naming.xml/LongVariable">
        <properties>
            <property name="maximum" value="30"/>
        </properties>
    </rule>
    <rule ref="rulesets/naming.xml/ShortMethodName"/>
    <rule ref="rulesets/naming.xml/ConstructorWithNameAsEnclosingClass"/>
    <rule ref="rulesets/naming.xml/ConstantNamingConventions"/>
    <rule ref="rulesets/naming.xml/BooleanGetMethodName"/>
    <rule ref="rulesets/unusedcode.xml"/>
</ruleset>
```

### PR DESCRIPTION
**Pull request by HcsOmot**
In order to use PHPMD in our dev work and ensure our code meets certain quality standards
As developers
We need to set up configuration for PHPMD
And run it automatically on CircleCI

Closes #124

### CODE REVIEW

File: phpmd.xml [hunk 1]
```diff
@@ -10,10 +10,9 @@
     <exclude-pattern>cache/</exclude-pattern>
     <exclude-pattern>SymfonyRequirements.php</exclude-pattern>
     <exclude-pattern>AppKernel.php</exclude-pattern>
-    <exclude-pattern>DoctrineMigrations/Version*.php</exclude-pattern>
+    <exclude-pattern>Migrations/Version*.php</exclude-pattern>
```
Comment (HcsOmot) [inline phpmd.xml]: @msvrtan we don't have a directory called `DoctrineMigrations`. Is this a remnant of a configuration for some version of Symfony prior to 4.x branch?

File: phpmd.xml [hunk 2]
```diff
@@ -10,10 +10,9 @@
...
     <exclude-pattern>DependencyInjection/Configuration.php</exclude-pattern>
     <exclude-pattern>DependencyInjection/*Extension.php</exclude-pattern>
-    <exclude-pattern>src/AppBundle/Controller/DefaultController.php</exclude-pattern>
```
Comment (HcsOmot) [inline phpmd.xml]: @msvrtan same here. If I'm not mistaken, this is from SF < v3.4, when Bundles were still used?

Comment (msvrtan) [reply phpmd.xml]: Feel free to remove

File: phpmd.xml (hunk 1 above)
Comment (msvrtan) [reply phpmd.xml]: Feel free to remove

**Pull request merged**